### PR TITLE
Fix `has_no_input_arg` check and rename it to `has_only_self_parameter`

### PR DESCRIPTION
**Pull request by 0xPoe**
Signed-off-by: Rustin-Liu <rustin.liu@gmail.com>

Fixes https://github.com/rust-lang/rust/pull/70643#discussion_r401571789.

### CODE REVIEW

File: src/librustc_typeck/check/demand.rs [hunk 1]
```diff
@@ -246,7 +246,10 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
     // This function checks if the method isn't static and takes other arguments than `self`.
```
Comment (Rustin-Liu) [inline src/librustc_typeck/check/demand.rs:246]: I am confusing about this comment, is this correct？

Comment (eddyb) [reply src/librustc_typeck/check/demand.rs:246]: It's a confusing comment. And the name of the function is also probably wrong. cc @estebank

Comment (estebank) [reply src/librustc_typeck/check/demand.rs:246]: This code seems to only be used for suggestions to convert a type in a whitelist, so I think the "has no input arg" should be read/rewritten as "check that this associated item can be called without any additional arguments, like <&str>::to_string(self)". Feel free to rewrite the docstring and method name.

Comment (Rustin-Liu) [reply src/librustc_typeck/check/demand.rs]: @estebank  @eddyb  Updated, PTAL~  Thanks!

**Pull request merged**